backend/api/auth.py

```
@@ -111,3 +142,64 @@ async def register(
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail=result["error"]
+            )
+            
+    except HTTPException:
+        # Re-raise HTTP exceptions (don't catch our own exceptions)
+        raise
+    except Exception as e:
+        # Handle unexpected database or system errors
+        # Log the actual error for debugging but don't expose internal details
+        print(f"Email confirmation error: {str(e)}")
```

Ruff (0.13.2)

Use explicit conversion flag

Replace with conversion flag

(RUF010)